src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,123 @@
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
 
 export const TodoList: React.FC = () => {
-  return (
-    <>
-      <p className="notification is-warning">
-        There are no todos matching current filter criteria
-      </p>
-
-      <table className="table is-narrow is-fullwidth">
-        <thead>
-          <tr>
-            <th>#</th>
-
-            <th>
-              <span className="icon">
-                <i className="fas fa-check" />
-              </span>
-            </th>
-
-            <th>Title</th>
-            <th> </th>
-          </tr>
-        </thead>
-
-        <tbody>
-          <tr data-cy="todo">
-            <td className="is-vcentered">1</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">delectus aut autem</p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">2</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">quis ut nam facilis et officia qui</p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
+  const todoList = useAppSelector(state => state.todos);
+  const openedTodo = useAppSelector(state => state.currentTodo);
+  const filter = useAppSelector(state => state.filter);
+  const dispatch = useAppDispatch();
+  const set = (value: Todo) => dispatch(actions.setTodo(value));
+
+  const filterTodos = () => {
+    switch (filter.status) {
+      case 'active':
+        return todoList.filter(todo => !todo.completed
+          && todo.title.toLowerCase()
+            .includes(filter.query.toLowerCase()));
+
+      case 'completed':
+        return todoList.filter(todo => todo.completed
+          && todo.title.toLowerCase()
+            .includes(filter.query.toLowerCase()));
+
+      case 'all':
+        return todoList.filter(todo => todo.title.toLowerCase()
+          .includes(filter.query.toLowerCase()));
+
+      default:
+        return todoList.filter(todo => todo.title.toLowerCase()
+          .includes(filter.query.toLowerCase()));
+    }
+  };
+
+  const filteredTodos
+  = useMemo(filterTodos, [filter, todoList]);
```

Comment:
You can write it in one line. And would be better to wrap the filterTodos into useCallback, and pass it to a dependency array. filter, todoList leave for a dependency for filterTodos
```suggestion
  const filteredTodos = useMemo(filterTodos, [filterTodos]);
```